src/client/routes/LabDetail/index.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import styled from 'styled-components';
+
+import {
+  AlertsActions,
+  AppSelectors,
+  TokensSelectors,
+  NetworkSelectors,
+  WalletSelectors,
+  LabsSelectors,
+  LabsActions,
+} from '@store';
+import { useAppDispatch, useAppSelector, useAppTranslation, useIsMounting } from '@hooks';
+import { ViewContainer, InfoCard } from '@components/app';
+import { SpinnerLoading, Button, Text } from '@components/common';
+import { isValidAddress } from '@utils';
+import { getConfig } from '@config';
+import { device } from '@themes/default';
+import { LabDetailPanels } from '@src/client/components/app/LabDetail';
+
+const BackButton = styled(Button)`
+  background-color: ${({ theme }) => theme.colors.surface};
+  color: ${({ theme }) => theme.colors.onSurfaceH2};
+`;
+
+const StyledInfoCard = styled(InfoCard)`
+  padding: 3rem;
+  margin: auto;
+`;
+
+const LabDetailView = styled(ViewContainer)`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+
+  @media ${device.mobile} {
+    ${StyledInfoCard} {
+      padding: 1rem;
+    }
+  }
+`;
+
+export interface LabDetailRouteParams {
+  labAddress: string;
+}
+
+export const LabDetail = () => {
+  const { t } = useAppTranslation(['common', 'labdetails']);
+  const dispatch = useAppDispatch();
+  const history = useHistory();
+  const location = useLocation();
+  const isMounting = useIsMounting();
+  const { NETWORK_SETTINGS } = getConfig();
+
+  const appStatus = useAppSelector(AppSelectors.selectAppStatus);
+  const selectedLab = useAppSelector(LabsSelectors.selectSelectedLab);
+  const labsStatus = useAppSelector(LabsSelectors.selectLabsStatus);
+  const tokensStatus = useAppSelector(TokensSelectors.selectWalletTokensStatus);
+  const currentNetwork = useAppSelector(NetworkSelectors.selectCurrentNetwork);
+  const walletIsConnected = useAppSelector(WalletSelectors.selectWalletIsConnected);
+  const walletName = useAppSelector(WalletSelectors.selectWallet);
+
+  const currentNetworkSettings = NETWORK_SETTINGS[currentNetwork];
+  const blockExplorerUrl = currentNetworkSettings.blockExplorerUrl;
+
+  const [firstTokensFetch, setFirstTokensFetch] = useState(false);
+  const [tokensInitialized, setTokensInitialized] = useState(false);
+
+  useEffect(() => {
+    const assetAddress: string | undefined = location.pathname.split('/')[2];
+    if (!assetAddress || !isValidAddress(assetAddress)) {
+      dispatch(AlertsActions.openAlert({ message: 'INVALID_ADDRESS', type: 'error' }));
+      history.push('/home');
+      return;
+    }
+    dispatch(LabsActions.setSelectedLabAddress({ labAddress: assetAddress }));
+
+    return () => {
+      dispatch(LabsActions.clearSelectedLabAndStatus());
+    };
+  }, [currentNetwork]);
+
+  useEffect(() => {
+    const loading = tokensStatus.loading;
+    if (loading && !firstTokensFetch) setFirstTokensFetch(true);
```

Comment:
I'm having a bit of a hard time following why we need the logic from firstTokensFetch